### PR TITLE
 Remove ScopeTarget and LoopIdResult

### PR DESCRIPTION
**Pull request by est31**
* Remove ScopeTarget in preparation of label-break-value (PR #50045)
* Replace LoopIdResult by Result which is possible now thanks to commit 8ac65af81f5f9cf6c5e2c2306705b50eed77cfb5 " Implement Encodable and Decodable for Result."

r? @eddyb

### CODE REVIEW

File: src/librustc_typeck/check/mod.rs [hunk 1]
```diff
@@ -3730,7 +3730,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
               tcx.mk_nil()
           }
           hir::ExprBreak(destination, ref expr_opt) => {
-              if let Some(target_id) = destination.target_id.opt_id() {
+              if let Some(target_id) = destination.target_id.ok() {
```
Comment (eddyb) [inline src/librustc_typeck/check/mod.rs]: This can pattern-match on `Ok` instead.

**Review by eddyb — CHANGES_REQUESTED** [review]

**Review by eddyb — APPROVED** [review]

**Comment by est31** [comment]
rebased. re-r? @eddyb

**Comment by kennytm** [comment]
@bors p=1

Needed to unblock #50045

**Pull request merged**